python/cuml/cuml/tests/test_linear_model.py

```diff
@@ -889,8 +817,11 @@ def test_logistic_regression_categorical_y(y_kind):
     assert res.dtype == "object"
 
 
-@pytest.mark.parametrize("train_dtype", [np.float32, np.float64])
-@pytest.mark.parametrize("test_dtype", [np.float64, np.float32])
+@given(train_dtype=dataset_dtypes(), test_dtype=dataset_dtypes())
+@example(train_dtype=np.float32, test_dtype=np.float32)
+@example(train_dtype=np.float32, test_dtype=np.float64)
+@example(train_dtype=np.float64, test_dtype=np.float32)
+@example(train_dtype=np.float64, test_dtype=np.float64)
```

Comment:
As a naive hypothesis user (aka not a lot of experience) I read this and thought "why do we still need the `given`? don't the examples cover all the possible combinations?"

Is this just a different way of writing what the `pytest.mark.parametrize` was doing? Why change?

Reply:
Yes, in this particular case they might actually cover all combinations. However, we might refine/expand our `dataset_types()` definition to include more types in the future (and at least until recently they included dtypes with varying endianness). I'm ok with leaving this as is even if it is currently a bit redundant.